Sources/PointFree/Rss.swift

```diff
@@ -0,0 +1,273 @@
+import Foundation
+import Html
+import Optics
+import Prelude
+
+struct RssChannel {
+  var copyright: String
+  var description: String
+  var image: Image
+  var itunes: Itunes?
+  var language: String
+  var link: String
+  var title: String
+
+  struct Image {
+    var link: String
+    var title: String
+    var url: String
+  }
+
+  struct Itunes {
+    var author: String
+    var categories: [Category]
+    var explicit: Bool
+    var keywords: [String]
+    var image: Image
+    var owner: Owner
+    var subtitle: String
+    var summary: String
+    var type: ChannelType
+
+    enum ChannelType: String {
+      case episodic
+      case serial
+    }
+
+    struct Image {
+      var href: String
+    }
+
+    struct Category {
+      var name: String
+      var subcategory: String
+    }
+
+    struct Owner {
+      var email: String
+      var name: String
+    }
+  }
+}
+
+struct RssItem {
+  var description: String
+  var dublinCore: DublinCore?
+  var enclosure: Enclosure?
+  var guid: String
+  var itunes: Itunes?
+  var link: String
+  var media: Media?
+  var pubDate: Date
+  var title: String
+
+  struct DublinCore {
+    var creators: [String]
+  }
+
+  struct Enclosure {
+    var length: Int
+    var type: String
+    var url: String
+  }
+
+  struct Itunes {
+    var author: String
+    var duration: Int
+    var episode: Int
+    var episodeType: EpisodeType
+    var explicit: Bool
+    var image: String
+    var subtitle: String
+    var summary: String
+    var season: Int
+    var title: String
+
+    enum EpisodeType: String {
+      case bonus
+      case full
+      case trailer
+    }
+  }
+
+  struct Media {
+    var content: Content
+    var title: String
+
+    struct Content {
+      var length: Int
+      var medium: String
+      var type: String
+      var url: String
+    }
+  }
+}
+
+func node(rssChannel channel: RssChannel, items: [RssItem]) -> Node {
+  let itunesNodes = channel.itunes.map(nodes(itunes:)) ?? []
+
+  return node(
+    "channel",
+    [
+      node("title", [text(channel.title)]),
+      node("link", [text(channel.link)]),
+      node("language", [text(channel.language)]),
+      node("description", [text(channel.description)]),
+      node("copyright", [text(channel.copyright)]),
+
+      node(
+        "image",
+        [
+          node("url", [text(channel.image.url)]),
+          node("title", [text(channel.image.title)]),
+          node("link", [text(channel.image.link)]),
+          ]
+      )
+      ]
+      + itunesNodes
+      + items.map(node(rssItem:))
+  )
+}
+
+func itunesRssFeedLayout<A>(_ view: View<A>) -> View<A> {
+  return View { a in
+    [
+      .text(unsafeUnencodedString("""
+      <?xml version="1.0" encoding="utf-8" ?>
+      """)),
+      node(
+        "rss",
+        [
+          attribute("xmlns:itunes", "http://www.itunes.com/dtds/podcast-1.0.dtd") as Attribute<Void>,
```

Comment:
I gotta do this `as Attribute<Void>` dance to get around the phantom type. is there a better way?